core/hydromet_stratified.py

```diff
@@ -557,4 +748,119 @@ def plot_max_potential_retention_cond_runoff(GEV_parameters_E: np.ndarray, PMP:
+    hyeto_graphs = np.where(hydro_events <1000, hydro_events, 1000).astype(int).tolist()
+    runoff_distros1 = {}
+    hydro_events = df_weights_rainfall.index.to_numpy().tolist()
+    prep_curves = pd.DataFrame(columns = hydro_events)
+    for event, hyetograph in zip(hydro_events, hyeto_graphs):
+        dist_df = get_hyeto_input_data_nrcs(nrcs_precip_table_dir, hyetograph, display_print)
+        dist_df['precip'] = dist_df['ratio']*df_weights_rainfall['Rainfall'].loc[event]
+        s = df_weights_rainfall['Avg. S'].loc[event]
+        ia = IA_24hr(s)
+        runoff_distros1[event] = excess_precip(dist_df,ia, s)
+        prep_curves[event] = dist_df['hyeto_input']
+    return prep_curves
+    
+def precip_to_runoff_h4(hydro_events:np.ndarray ,atlas14_precip_table_dir: pl.WindowsPath,
+                     precip_data: pd.DataFrame,df_weights_rainfall: pd.DataFrame, display_print = False) -> pd.DataFrame:
+    """Takes the events, precipitation data, atlas 14 temporal distribution, selected CNs and applies the 
+    CN reduction method to obtain a runoff curve for each recurrence interval. 
+    """  
+    Atlas14_hyetographs = ['q1', 'q2', 'q3', 'q4']
+    hyeto_graphs = np.where(hydro_events <1000, hydro_events, 1000).astype(int).tolist()
+    hydro_events = df_weights_rainfall.index.to_numpy().tolist()
+    column_names = []
+    
+    for event in hydro_events:
+        for hyetograph in Atlas14_hyetographs:
+            column_names.append(event+'_'+hyetograph)
+            
+    prep_curves = pd.DataFrame(columns = column_names)
+    prep_weights = pd.DataFrame(index = column_names, columns= ['Event Weight'])
+    
+    for event in hydro_events:
+        for  hyetograph in  Atlas14_hyetographs:
+            dist_df, weight_df = get_hyeto_input_data_atlas(atlas14_precip_table_dir, hyetograph, display_print)
+            dist_df['precip'] = dist_df[hyetograph]*df_weights_rainfall['Rainfall'].loc[event]
+            s = df_weights_rainfall['Avg. S'].loc[event]
+            ia = IA_24hr(s)
+            dist_df = excess_precip(dist_df,ia, s)
+            prep_curves[event+'_'+hyetograph] = dist_df['hyeto_input']
+            prep_weights['Event Weight'].loc[event+'_'+hyetograph] = df_weights_rainfall['Event Weight'][event]*weight_df['weight'][hyetograph]
+    return prep_curves, prep_weights
+
+def extend_time(prep_curves: pd.DataFrame,time_extend: float,time_step: float) -> pd.DataFrame:
+    """extends the hyetograph by a select period of time. the timestep is the spacing between
+       simulation intervals (typically 0.1 or 0.5 hours)
+    """
+    extend_curves = prep_curves.loc[0.0:time_extend]*0
+    extend_curves.index = extend_curves.index+(24+time_step)
+    return prep_curves.append(extend_curves).rename_axis('hours')
+
+def S_24hr(CN: int) -> float:
+    '''Calculates the potential maximum retention after runoff begins (S), in 
+       inches.
+    '''
+    return (1000-10*CN)/CN
+
+
+def IA_24hr(s24: float) -> float:
```

Comment:
This is already in hydromet.py, can delete function

Reply:
Removed